src/Shared/src/ActivatorUtilities/ActivatorUtilities.cs

```diff
@@ -327,18 +329,16 @@ private static bool TryCreateParameterMap(ParameterInfo[] constructorParameters,
             return true;
         }
 
-        private class ConstructorMatcher
+        private struct ConstructorMatcher
         {
             private readonly ConstructorInfo _constructor;
             private readonly ParameterInfo[] _parameters;
             private readonly object[] _parameterValues;
-            private readonly bool[] _parameterValuesSet;
```

Comment:
`GetTypeInfo().IsAssignableFrom(null)` returns `false` (around line 357), so I don't see why we need `parameterValuesSet`.